src/fluree/db/json_ld/commit.cljc

```diff
@@ -350,6 +351,20 @@
           db**           (assoc-in db* [:ecount const/$_shard] @last-sid)]
       (add-commit-flakes-to-db db** flakes))))
 
+(defn link-context-to-commit
+  "Takes a commit with an embedded :context and pulls it out, saves it to
+  storage separately (content-addressed), and puts the address of that data
+  back into the commit under the :context key."
+  [{:keys [conn] :as ledger} commit]
+  (go-try
+    (let [context (get commit (keyword const/iri-default-context))
+          stringify? (-> context keys first keyword?) ; (too?) simple check if we need to stringify the keys before storing
+          context-str (if stringify?
+                        (walk/stringify-keys context)
```

Comment:
We have a simpler `util/stringify-keys` that I think would work better and fits this use case. Walk not only brings in a whole new namespace + its backing functions for a simple task, it also is more generic to apply to many data structures - where our internal function is a simple reduce designed for simple maps just like this.